### DISCUSSION TITLE
How to get contributor history and contributors in branch

### DISCUSSION THREAD

**Discussion by salah-elraies** [community/community#60647] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I'm building a website that has a feature that needs an API that gets all contributors to a branch with their data and an API that gets all branches that a contributor worked on.
I've been digging in the docs but can't seem to find what I'm looking for.

**Comment by nirav-gajera** [comment]
Get the Contributors: You can use the Git command git log to retrieve the commit history and extract the contributors' names.

`git log --format='%aN' <branch_name> | sort -u
`

Get Contributor History: To get the history of a specific contributor, you can use the git log command with the --author flag to filter the commits by the contributor's name

`git log --author="<contributor_name>"
`

Get Branches for a Contributor: To find the branches that a contributor has worked on, you can iterate over all branches in the repository and check for commits made by the contributor.

`git branch -r --contains <contributor_name>
`

**Reply by salah-elraies** [reply]
Thank you so much.
I'd really appreciate it if you can tell me how to implement these command in a website

**Reply by nirav-gajera** [reply, marked as answer]
1. Set up a Server: Choose a server-side technology stack such as Node.js with Express, Python with Flask/Django, or any other language/framework you're comfortable with. This will serve as the backend for your website.

2. Execute Git Commands: In your server-side code, use the appropriate functions or libraries to execute the Git commands programmatically. For example, you can use child_process module in Node.js or subprocess module in Python to execute shell commands.

3. Capture Command Output: After executing the Git commands, capture the output from the command execution. You can capture the output as a string or parse it into structured data as per your requirements. Make sure to handle any error cases and provide appropriate error handling in your code.

4. Expose APIs: Create API endpoints in your server-side code to handle the requests from your web application. For example, you can create an API endpoint like /contributors that internally executes the git log command and returns the list of contributors as a JSON response.

5. Connect Frontend and Backend: In your web application's frontend code (HTML, CSS, and JavaScript), make AJAX requests to the API endpoints you created in the backend. You can use JavaScript libraries like Axios or the built-in Fetch API to make these HTTP requests from the client-side to your server-side.

6. Process and Display Results: Receive the API responses in your frontend code and process the data to display it on your website. You can use JavaScript frameworks like React, Vue.js, or plain JavaScript to manipulate and render the received data.

**Reply by salah-elraies** [reply]
thank you, this is a real live saver.
